runtime/src/tx_utils.rs

```diff
@@ -0,0 +1,75 @@
+use std::ops::Index;
```

Comment:
nit: I'd rather tx be expanded in filenames/function names to avoid ambiguity.   So this file would be called `transaction_utils.rs` instead

Reply:
Done